### PR TITLE
Add integ test code coverage support and cross module code coverage

### PR DESCRIPTION
**Pull request by zoewangg**
## Description
Add integ test code coverage support and cross module code coverage

a new module `tests-coverage-reporting` is added to aggregate the testing reports so we can see the cross module tests coverage

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 [![Build Status](https://travis-ci.org/aws/aws-sdk-java-v2.svg?branch=master)](https://travis-ci.org/aws/aws-sdk-java-v2) ![Build Status](https://codebuild.us-west-2.amazonaws.com/badges?uuid=eyJlbmNyeXB0ZWREYXRhIjoiTFJSRXBBN1hkU1ZEQzZ4M1hoaWlFUExuNER3WjNpVllSQ09Qam1YdFlTSDNTd3RpZzNia3F0VkJRUTBwZlQwR1BEelpSV2dWVnp4YTBCOFZKRzRUR004PSIsIml2UGFyYW1ldGVyU3BlYyI6ImdHdEp1UHhKckpDRmhmQU4iLCJtYXRlcmlhbFNldFNlcmlhbCI6MX0%3D&branch=master)
 [![Maven](https://img.shields.io/maven-central/v/software.amazon.awssdk/s3.svg?label=Maven)](https://search.maven.org/search?q=g:%22software.amazon.awssdk%22%20AND%20a:%22s3%22)
 [![Gitter](https://badges.gitter.im/aws/aws-sdk-java-v2.svg)](https://gitter.im/aws/aws-sdk-java-v2?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge) 
+[![codecov](https://codecov.io/gh/aws/aws-sdk-java-v2/branch/master/graph/badge.svg)](https://codecov.io/gh/aws/aws-sdk-java-v2)
```
Comment (millems) [inline README.md]: What's this needed for?

File: test/tests-coverage-reporting/pom.xml [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+        <artifactId>aws-sdk-java-pom</artifactId>
+        <groupId>software.amazon.awssdk</groupId>
+        <version>2.5.40-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>tests-coverage-reporting</artifactId>
+    <name>AWS Java SDK :: Test :: Tests Coverage Reporting</name>
+    <description>Uses to aggregate integration and unit tests coverage reports
```
Comment (millems) [inline test/tests-coverage-reporting/pom.xml]: Nit: Used*

File: test/tests-coverage-reporting/pom.xml [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        </dependency>
+        <dependency>
+            <artifactId>codegen-generated-classes-test</artifactId>
+            <groupId>software.amazon.awssdk</groupId>
+            <version>${awsjavasdk.version}</version>
+        </dependency>
+
+        <!-- Need to explicitly add service modules to aggregate the tests coverage
+        and a few services that we know with more tests should be sufficient
+        -->
```
Comment (millems) [inline test/tests-coverage-reporting/pom.xml]: Can we add the bom or aggregate package as a direct dependency or something?

File: test/tests-coverage-reporting/pom.xml [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+                    </ignoredUnusedDeclaredDependencies>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
+                <version>${jacoco-maven-plugin.version}</version>
+                <executions>
+                    <execution>
+                        <id>report-aggregate</id>
```
Comment (millems) [inline test/tests-coverage-reporting/pom.xml]: Can this not be done in the root pom or some different package that already depends on everything?

File: README.md (hunk 1 above)
Comment (zoewangg) [reply README.md]: Just showing off the tests coverage.

File: test/tests-coverage-reporting/pom.xml (hunk 3 above)
Comment (zoewangg) [reply test/tests-coverage-reporting/pom.xml]: Nope, I tried and it didn't work. Dependencies have to be added explicitly.

File: test/tests-coverage-reporting/pom.xml (hunk 4 above)
Comment (zoewangg) [reply test/tests-coverage-reporting/pom.xml]: I tried root pom and it didn't work. We could use `aws-java-sdk` module but it doesn't depend on the test modules such as `protocol-tests` and we definitely want to include them.

File: README.md (hunk 1 above)
Comment (millems) [reply README.md]: Ah, cool, makes sense. I thought this was a link at the bottom, not a badge at the top.

File: test/tests-coverage-reporting/pom.xml (hunk 3 above)
Comment (millems) [reply test/tests-coverage-reporting/pom.xml]: Weird. I suppose we can iterate on it if it becomes a maintenance problem.

File: test/tests-coverage-reporting/pom.xml (hunk 2 above)
Comment (zoewangg) [reply test/tests-coverage-reporting/pom.xml]: Yup, fixed

**Pull request merged**